src/APIs/FlagsAPI.ts

```diff
@@ -0,0 +1,56 @@
+import compact from 'lodash/compact';
+import assign from 'lodash/assign';
+import omit from 'lodash/omit';
+
+interface Word {
+  id?: string;
+  stems?: Array<string | { id?: string }>;
+  relatedTerms?: Array<string | { id?: string }>;
+  examples?: any; // Update the type of `examples` accordingly
+  dialects?: any; // Update the type of `dialects` accordingly
+}
+
+interface Data {
+  words: Word[];
+  contentLength: number;
+}
+
+interface Flags {
+  examples?: boolean;
+  dialects?: boolean;
+  resolve?: boolean;
+}
+
+interface HandleWordFlagsParams {
+  data: Data;
+  flags: Flags;
+}
+
+export const handleWordFlags = ({
+  data: { words, contentLength },
+  flags: { examples, dialects, resolve },
+}: HandleWordFlagsParams) => {
+  const updatedWords = compact(
+    words.map((word) => {
+      let updatedWord = assign({}, word);
```

Comment:
Is there a reason why the Loadash is used instead of the native `Object.assign` method?